astropy/io/fits/tests/test_convenience.py

```diff
@@ -255,6 +270,41 @@ def test_tabledump(self):
         fits.tabledump(self.temp('tb.fits'), datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
 
+        # test with datafile and cdfile
+        datafile = self.temp('data.txt')
+        cdfile = self.temp('coldefs.txt')
+        fits.tabledump(self.temp('tb.fits'), datafile, cdfile)
+        assert os.path.isfile(datafile)
+        with open(datafile) as data:
+            assert data.read() == datastr
+        with open(cdfile) as coldefs:
+            assert coldefs.read() == cdstr
+
+    @pytest.mark.parametrize('tablename', ['table.fits', 'tb.fits'])
+    def test_dump_load_round_trip(self, tablename):
+        """
+        A simple test of the dump/load methods; dump the data, column, and
+        header files and try to reload the table from them.
+        """
+
+        # copy fits file to the temp directory
+        self.copy_file(tablename)
+
+        datafile = self.temp('data.txt')
+        cdfile = self.temp('coldefs.txt')
+        hfile = self.temp('header.txt')
+        fits.tabledump(self.temp(tablename), datafile, cdfile, hfile)
+
+        new_tbhdu = fits.tableload(datafile, cdfile, hfile)
+
+        with fits.open(self.temp(tablename)) as hdul:
+            # Double check that the headers are equivalent
+            for card in hdul[1].header.cards:
+                # Skip unloaded keywords (should this be fixed?)
+                if card.keyword[0:5] not in ['TFORM', 'TDISP', 'TNULL', 'TZERO', 'TSCAL']:
+                    new_card = new_tbhdu.header.cards[card.keyword]
+                    assert card.value == new_card.value
```

Comment:
Maybe one option would be to compare the coldefs attributes instead, with something like this (that could go in a new helper function):
```
from astropy.io.fits.column import ColumnAttribute

assert hdu.columns.names == new_tbhdu.columns.names
attrs = [k for k, v in fits.Column.__dict__.items() if isinstance(v, ColumnAttribute)]

for name in hdu.columns.names:
    col = hdu.columns[name]
    new_col = new_tbhdu.columns[name]
    for attr in attrs:
        if getattr(col, attr) and getattr(new_col, attr):
             assert getattr(col, attr) == getattr(new_col, attr)
```